File: clojure/two-fer/src/two_fer.clj
(ns two-fer)

(defn two-fer
  "Returns the \"One For\" string with the name passed in, otherwise \"you\"."
  ([]
   {:pre []
    :post [(string? %)]}
   (two-fer "you"))
  ([name]
   {:pre [(string? name)]
    :post [(string? %)]}
   (str "One for " name ", one for me.")))

File: java/two-fer/src/main/java/Twofer.java
public class Twofer {
    // public String twofer(String name) {
    //     return "One for " + (name == null ? "you" : name) + ", one for me.";
    // }
    
    public String twofer(String name) {
        if (name == null){
            return "One for you, one for me.";
        } else {
            return "One for " + name + ", one for me.";
        }
    }

}